framework/JobHandler.py

```diff
@@ -402,9 +402,11 @@ def availability(self, client=False):
     ## should also be addressed, but at least we can prevent it on this end since
     ## the main thread's issue may be legitimate.
     if client:
-      availability = self.maxQueueSize - len(self.__clientQueue) - len(self.__finished)
+      #availability = self.maxQueueSize - len(self.__clientQueue) - len(self.__finished)
+      availability = self.__clientRunning.count(None) - len(self.__clientQueue) - len(self.__finished)
     else:
-      availability = self.maxQueueSize - len(self.__queue) - len(self.__finished)
+      #availability = self.maxQueueSize - len(self.__queue) - len(self.__finished)
+      availability = self.__running.count(None) - len(self.__queue) - len(self.__finished)
```

Comment:
If we make this change, we would probably want to remove the `self.maxQueueSize` variable. I think I would rather keep that variable (but I am willing to hear counter arguments), and instead set it to the size of `self.__running` by default. Eventually, we may want to expose this to the user, as there are times when it is beneficial to have more jobs "prepared" to run and other times where having too many "prepared" runs can actually gum up the memory usage. Thoughts?

Reply:
I can set ```self.maxQueueSize = self.__running``` for now, but there  will be a potential issue for this implementation. What I tried to do is 1) "online rom training", 2) Automatic online model selection between the model and rom. If we queue two many jobs, we will loose the efficiency because we are running the model until the all the queued jobs are finished instead of rom even if the rom is ready to be used to replace the model.

Reply:
Hmm, I see. So, even though these jobs are queued with the `HybridModel` which in theory can select whether to run the model or the ROM, the `HybridModel` is frozen.  This means if we queue a bunch of jobs, then it will likely run the expensive model for those jobs since  the updates from other runs are not known by the state of the `HybridModel` when it was frozen.

Reply:
This is a valid point, but I wonder how we can marry this with the other workflow for non-hybrid models? We could set self.maxQueueSize to None by default and in that event, then we will use the line you have written above?

Otherwise, I would say we could try the methodology you have and see if we see any degradation in performance on some heavier tests (that do not use the hybrid model). When I initially put in this feature, I was trying to prevent the issue I described, but I don't recall if I had any hard evidence to support that this was happening in RAVEN in practice. It could be that your method works fine in both cases.

Reply:
You got my point. Maybe we can let the user to decide the value for self.maxQueueSize, and by default to None. We can clearly describe the situations for the user when they need to increase the value for self.maxQueueSize in the manual or the user guide.